examples/step-82/step-82.cc

```diff
@@ -0,0 +1,1500 @@
+  template <int dim>
+  void BiLaplacianLDGLift<dim>::compute_discrete_hessians(
+    const typename DoFHandler<dim>::active_cell_iterator &cell,
+    const typename DoFHandler<dim>::active_cell_iterator &cell_lift,
+    std::vector<std::vector<Tensor<2, dim>>> &            discrete_hessians,
+    std::vector<std::vector<std::vector<Tensor<2, dim>>>>
+      &discrete_hessians_neigh)
+  {
+    QGauss<dim>     quad(fe.degree + 1);
+    QGauss<dim - 1> quad_face(fe.degree + 1);
+
+    const unsigned int n_q_points      = quad.size();
+    const unsigned int n_q_points_face = quad_face.size();
+
+    // The information we need from the basis functions of
+    // $\mathbb{V}_h$: <code>fe_values</code> is needed to add
+    // the broken Hessian part of the discrete Hessian, while
+    // <code>fe_face</code> and <code>fe_face_neighbor</code>
+    // are used to compute the right-hand sides for the local
+    // problems.
+    FEValues<dim> fe_values(fe, quad, update_hessians | update_JxW_values);
+
+    FEFaceValues<dim> fe_face(
+      fe, quad_face, update_values | update_gradients | update_normal_vectors);
+
+    FEFaceValues<dim> fe_face_neighbor(
+      fe, quad_face, update_values | update_gradients | update_normal_vectors);
+
+    const unsigned int n_dofs = fe_values.dofs_per_cell;
+
+    // The information needed from the basis functions
+    // of the finite element space for the lifting terms:
+    // <code>fe_values_lift</code> is used for the (local)
+    // mass matrix (see $\boldsymbol{M}_c$ in the introduction),
+    // while <code>fe_face_lift</code> is used to compute the
+    // right-hand sides (see $\boldsymbol{G}_c$ for $b_e$).
+    FEValues<dim> fe_values_lift(fe_lift,
+                                 quad,
+                                 update_values | update_JxW_values);
+
+    FEFaceValues<dim> fe_face_lift(
+      fe_lift, quad_face, update_values | update_gradients | update_JxW_values);
+
+    const FEValuesExtractors::Tensor<2> tau_ext(0);
+
+    const unsigned int n_dofs_lift = fe_values_lift.dofs_per_cell;
+    FullMatrix<double> local_matrix_lift(n_dofs_lift, n_dofs_lift);
+
+    Vector<double> local_rhs_re(n_dofs_lift), local_rhs_be(n_dofs_lift),
+      coeffs_re(n_dofs_lift), coeffs_be(n_dofs_lift), coeffs_tmp(n_dofs_lift);
+
+    SolverControl solver_control(1000, 1e-12);
+    SolverCG<>    solver(solver_control);
+
+    double factor_avg; // 0.5 for interior faces, 1.0 for boundary faces
+
+    fe_values.reinit(cell);
+    fe_values_lift.reinit(cell_lift);
+
+    // We start by assembling the (local) mass matrix used for the computation
+    // of the lifting terms $r_e$ and $b_e$.
+    assemble_local_matrix(fe_values_lift, n_q_points, local_matrix_lift);
+
+    for (unsigned int i = 0; i < n_dofs; ++i)
+      for (unsigned int q = 0; q < n_q_points; ++q)
+        {
+          discrete_hessians[i][q] = 0;
+
+          for (unsigned int face_no = 0;
+               face_no < GeometryInfo<dim>::faces_per_cell;
```

Comment:
```suggestion
               face_no < discrete_hessians_neigh.size();
```

One `GeometryInfo` less :)

Reply:
Thanks!